test/cmdlineTests/~bytecode_equivalence_multiple_sources/test.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import subprocess
+from pathlib import Path
+from shutil import which
+from tempfile import mkdtemp
+
+PROJECT_ROOT = Path(__file__).parents[3]
+CMDLINE_TEST_DIR = PROJECT_ROOT / "test/cmdlineTests"
+
+sys.path.insert(0, str(PROJECT_ROOT / "scripts"))
+#pylint: disable=wrong-import-position
+from bytecodecompare.prepare_report import FileReport
+from bytecodecompare.prepare_report import parse_cli_output
+from splitSources import split_sources
+
+
+class FileMismatchError(Exception):
+    def __str__(self):
+        return f"Git diff mismatch:\n {self.args[0]}"
+
+
+def diff(file_a: Path, file_b: Path):
+    """
+    Returns with no output if there are no differences between the two inputs.
+    Otherwise (i.e. if the process exits with a non-zero exit code) throws FileMismatchError.
+    """
+
+    if which("git") is None:
+        raise RuntimeError("git not found.")
+
+    try:
+        subprocess.check_output(
+            f"""\
+            git diff \
+                --color \
+                --word-diff=plain \
+                --word-diff-regex=. \
+                --ignore-space-change \
+                --ignore-blank-lines \
+                {file_a} {file_b}
+            """,
+            encoding="utf8",
+            shell=True
+        )
+    except subprocess.CalledProcessError as e:
+        raise FileMismatchError(e.output) from e
+
+
+def run_solc(report_file_path: Path, solc_binary_path: Path, args = None) -> FileReport:
+    if args is None:
+        args = []
+    output = subprocess.check_output(
+        [solc_binary_path] + args,
+        shell=False,
+        encoding="utf8"
+    )
+    return parse_cli_output(report_file_path, output)
+
+
+def save_bytecode_report(contract: str, bytecode_path: Path, reports: FileReport):
+    for report in reports.contract_reports:
+        if report.contract_name == contract:
+            with open(bytecode_path, 'w', encoding='utf8', newline='') as f:
+                f.write(report.bytecode if report.bytecode is not None else "<NO BYTECODE>")
+
+
+def test_bytecode_equivalence(tmp_dir: Path):
+    previous_dir = os.getcwd()
+    os.chdir(tmp_dir)
+
+    source_file_path = CMDLINE_TEST_DIR / "~bytecode_equivalence_multiple_sources/inputs.sol"
```

Comment:
It would be best if we did not have to hard-code the test name in it. It's always annoying when you move it. Should be easy to do - the file is in the same dir after all.